packages/utils/node-resolver-core/src/NodeResolver.js

```diff
@@ -602,7 +653,16 @@ export default class NodeResolver {
       if (isExplicitNode) {
         filename = filename.substr(5);
       }
-      return {filePath: builtins[filename] || empty};
+
+      // By default, exclude node builtins from libraries unless explicitly opted in.
```

Comment:
Is this a good idea? Alternatively we could have some separate config for it, but it seems like when building libraries you'd usually want to exclude these...

Reply:
Since it is ideal to resolve node builtins polyfill with the bundler of the library user, I feel we do not want to include node builtins if a bundling target is a library working in the browser. On the other hand, I think some library author wants to host their library on `elm.sh` etc. and use it as pure ESModules from the browser. In that case, the bundle of the library should include node builtins.


So, I think it is good to make it the default behavior, but I think also that an emergency hatch should be prepared as a config or flag for such use cases. (it would be simple to judge that with whether or not the output is `esmodule`, but in reality, it shouldn't be so difficult ….)

I'm sorry if I overlooked the context and said something strange